agent/agent_endpoint.go

```diff
@@ -170,10 +170,11 @@ func (s *HTTPServer) AgentChecks(resp http.ResponseWriter, req *http.Request) (i
 	}
 
 	// Use empty list instead of nil
-	// checks needs to be a deep copy for this not be racy
-	for _, c := range checks {
+	for id, c := range checks {
 		if c.ServiceTags == nil {
-			c.ServiceTags = make([]string, 0)
+			clone := *c
```

Comment:
Same issue in https://github.com/hashicorp/consul/pull/3867/files#diff-3b7c7efd1037ec5aaf67f1c2e349cf82R149

Reply:
I don't know what issue you are referring to with this comment.

Reply:
You fixed it in https://github.com/hashicorp/consul/pull/3867/commits/c2a59f1e6c377e1d99c69809236f1de22e996219, around line 150.